### PR TITLE
modularize the wing model

### PR DESCRIPTION
**Pull request by mjburton11**
Allows easy use of wing model for adding components and use as vtail or htail

### CODE REVIEW

File: gpkitmodels/GP/aircraft/wing/wing.py [hunk 1]
```diff
@@ -29,9 +29,9 @@ def setup(self, N):
                            lambda c: 2.0/3.0*(1+c[lam]+c[lam]**2)/(1+c[lam]),
                            "-", "non-dim MAC")
         with Vectorize(N):
+            eta = Variable("\\eta", "-", "(2y/b)")
             cbar = Variable("\\bar{c}", "-",
```
Comment (mjburton11) [inline gpkitmodels/GP/aircraft/wing/wing.py]: @bqpd I want this to be, but had problems doing so

```python
eta = Variable("\\eta", np.linspace(0, 1, N), "-", "(2y/b)")
cbar = Variable("\\bar{c}", lambda c: 2.0/(1 + c[lam])*(1 + (c[lam] - 1)*c[eta]), "-", "non-dim chord length"
```

Comment (mjburton11) [reply gpkitmodels/GP/aircraft/wing/wing.py]: @bqpd, also would it be possible to do this:
```python
def return_cmac(lam):
    return 2.0/3.0*(1+c[lam]+c[lam]**2)/(1+c[lam])
cbarmac = Variable("\\bar{c}_{MAC}", return_cmac(lam), "-", "non-dim MAC")
```

**Comment by mjburton11** [comment]
test models please

**Comment by mjburton11** [comment]
@bqpd tests have been in like this for a while (Waiting for status to be reported).  Do you know what is going on?

**Comment by mjburton11** [comment]
test models please

**Comment by bqpd** [comment]
I think the merge conflicts have to be fixed before you can test

**Comment by mjburton11** [comment]
test models please

**Comment by mjburton11** [comment]
@bqpd dah, still in this waiting state...

**Comment by mjburton11** [comment]
test models please

**Comment by mjburton11** [comment]
@galbramc do you know why the status is "waiting for status to be reported"? It's been that way for some time, despite attempts to trigger another test using "test models please"

**Comment by galbramc** [comment]
hmm, no let me check jenkins (you can also always check the progress here https://acdl.mit.edu/csi/)

**Comment by galbramc** [comment]
test models please

**Comment by mjburton11** [comment]
I was looking there, but couldn't figure out what had happened.

**Comment by galbramc** [comment]
did this repo get renamed?

**Comment by mjburton11** [comment]
yeah I think it was library a day or two ago and now its gplibrary

**Comment by galbramc** [comment]
Ok. I don't actually see a test on jenkins for this repo. Did @bqpd try to set one up?

**Comment by mjburton11** [comment]
Yeah we had it testing thursday I think

**Comment by galbramc** [comment]
Is this repo supposed to be tested just like SPAircraft and turbofan for example?

**Comment by mjburton11** [comment]
No it's supposed to set up more like gpkit, testing the "research repo" like SPaircraft and turbofan and gassolar and jho

**Comment by galbramc** [comment]
Oh is it the gpkit_commons_PR_Models and gpkit_commons_Push_Models tests?

**Comment by mjburton11** [comment]
I believe its gpkit_commons_PR_Models

**Comment by galbramc** [comment]
test models please

**Comment by mjburton11** [comment]
that's seemed to work!

**Comment by galbramc** [comment]
yeah when you rename a repo you have to update the names in jenkins

**Comment by galbramc** [comment]
How is this repo related to models? Do you really want to say "test models" to run tests here? would you prefer "test this"?

**Comment by mjburton11** [comment]
ahh, ok thanks!

**Comment by galbramc** [comment]
I setup "test models" with gpkit so you could run all the tests in all the model repos when you are making changes to gpkit. Does that somehow apply to this repo as well?

**Comment by mjburton11** [comment]
I don't know.  @bqpd is the one who knows that

**Comment by mjburton11** [comment]
test models please

**Comment by mjburton11** [comment]
test models please

**Comment by galbramc** [comment]
your tests are chaotic. You get different results every time you run them without making changes. That does not make them particularly useful tests.

**Comment by mjburton11** [comment]
It's a problem with the turbofan repo, which my changes don't depend on... not sure how to best fix that.

**Comment by mjburton11** [comment]
test models please

**Comment by bqpd** [comment]
derp, sorry, forgot the renames break jenkins

File: gpkitmodels/GP/aircraft/wing/wing.py [hunk 2]
```diff
@@ -25,52 +24,40 @@ def setup(self, N=5, lam=0.5, spar="CapSpar", hollow=False):
...
+            eta = Variable("\\eta", "-", "(2y/b)")
+            cbar = Variable("\\bar{c}", "-",
                             "normalized chord at mid element")
             eta = Variable("\\eta", "-", "(2y/b)")
         with Vectorize(N-1):
-            cbave = Variable("\\bar{c}_{ave}", (cb[1:]+cb[:-1])/2, "-",
+            cbave = Variable("\\bar{c}_{ave}", "-",
                              "normalized mid section chord")
             cave = Variable("c_{ave}", "ft", "mid section chord")
+            deta = Variable("d\\eta", "-", "\\Detla (2y/b)")
```
Comment (bqpd) [inline gpkitmodels/GP/aircraft/wing/wing.py]: is the description right on this variable?

File: gpkitmodels/GP/aircraft/tail/horizontal_tail.py [hunk 3]
```diff
@@ -1,48 +1,30 @@
 " horizontal tail "
 import numpy as np
 from gpkit import Model, Variable
-from tail_aero import TailAero
+from .tail_aero import TailAero
+from gpkitmodels.GP.aircraft.wing.wing import Wing
+from gpkitmodels.GP.aircraft.wing.constant_taper_chord import c_bar
```
Comment (bqpd) [inline gpkitmodels/GP/aircraft/tail/horizontal_tail.py]: are all these new imports actually used in this file?

File: gpkitmodels/GP/aircraft/tail/vertical_tail.py [hunk 4]
```diff
@@ -1,42 +1,27 @@
 " vertical tail "
 from gpkit import Model, Variable
-from tail_aero import TailAero
+from .tail_aero import TailAero
+from gpkitmodels.GP.aircraft.wing.wing import Wing
+from gpkitmodels.GP.aircraft.wing.constant_taper_chord import c_bar
+from gpkitmodels.GP.aircraft.wing.wing_interior import WingInterior
+from gpkitmodels.GP.aircraft.wing.wing_skin import WingSkin
```
Comment (bqpd) [inline gpkitmodels/GP/aircraft/tail/vertical_tail.py]: are these new imports actually used?

File: gpkitmodels/GP/aircraft/wing/constant_taper_chord.py [hunk 5]
```diff
@@ -6,4 +6,5 @@ def c_bar(lam, N):
     eta = np.linspace(0, 1, N)
     c = 2/(1+lam)*(1+(lam-1)*eta)
     cbarmac = 2./3*(1+lam+lam**2)/(1+lam)
-    return c, eta, cbarmac
+    deta = np.diff(eta)
+    return c, eta, deta, cbarmac
```
Comment (bqpd) [inline gpkitmodels/GP/aircraft/wing/constant_taper_chord.py]: add a blank line to the end of the file

**Comment by mjburton11** [comment]
test models please

**Pull request merged**